src/supermarket/app/controllers/cookbooks_controller.rb

```diff
@@ -308,4 +309,17 @@ def apply_filters
       @cookbooks = @cookbooks.filter_badges(params[:badges])
     end
   end
+
+  def classify_metrics(metric_results)
```

Comment:
Seeds of ideas.

`cbv = Cookbook.last.latest_cookbook_version`

### use `includes(:quality_metrics)` to have results preloaded with associated metrics

Prevents the N+1 queries for the `.each` loop interrogating for whether a metric is admin only.

```
> cbv.metric_results.includes(:quality_metric)
  MetricResult Load (1.0ms)  SELECT "metric_results".* FROM "metric_results" WHERE "metric_results"."cookbook_version_id" = $1  [["cookbook_version_id", 27]]
  QualityMetric Load (0.7ms)  SELECT "quality_metrics".* FROM "quality_metrics" WHERE "quality_metrics"."id" IN (1, 3, 2)
=> [#<MetricResult:0x007fa883e1bd28
  id: 1,
  cookbook_version_id: 27,
  quality_metric_id: 1,
  failure: true,
  feedback:
   "FC064: Ensure issues_url is set in metadata: /var/folders/py/gk9z_67s0mjbg0w5m3xl_w180000gn/T/f78ce2d0ed32b58973518623/dragons/metadata.rb:1\nFC065: Ensure source_url is set in metadata: /var/folders/py/gk9z_67s0mjbg0w5m3xl_w180000gn/T/f78ce2d0ed32b58973518623/dragons/metadata.rb:1\nRun with Foodcritic Version 7.0.1 with tags ~FC031 ~FC045 and failure tags any",
  created_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00,
  updated_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00>,
 #<MetricResult:0x007fa883e1bb20
  id: 2,
  cookbook_version_id: 27,
  quality_metric_id: 3,
  failure: false,
  feedback: "",
  created_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00,
  updated_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00>,
 #<MetricResult:0x007fa883e1b850
  id: 3,
  cookbook_version_id: 27,
  quality_metric_id: 2,
  failure: true,
  feedback: "Failure: Cookbook has 0 collaborators. A cookbook must have at least 2 collaborators to pass this metric.",
  created_at: Tue, 15 Nov 2016 16:44:14 UTC +00:00,
  updated_at: Tue, 15 Nov 2016 16:44:14 UTC +00:00>]
```

### use `includes` plus a delegated `admin_only` attribute to group returned metrics by admin-onliness

The `nil` key's array of MetricResults are the results viewable by everyone.
The `true` key's array of MetricResults are the admin-only ones.

```
[21] pry(main)> class MetricResult; delegate :admin_only, to: :quality_metric ; end
=> [:admin_only]
[22] pry(main)> cbv.metric_results.includes(:quality_metric).group_by(&:admin_only)
  MetricResult Load (0.8ms)  SELECT "metric_results".* FROM "metric_results" WHERE "metric_results"."cookbook_version_id" = $1  [["cookbook_version_id", 27]]
  QualityMetric Load (0.7ms)  SELECT "quality_metrics".* FROM "quality_metrics" WHERE "quality_metrics"."id" IN (1, 3, 2)
=> {nil=>
  [#<MetricResult:0x007fa8825eeac0
    id: 1,
    cookbook_version_id: 27,
    quality_metric_id: 1,
    failure: true,
    feedback:
     "FC064: Ensure issues_url is set in metadata: /var/folders/py/gk9z_67s0mjbg0w5m3xl_w180000gn/T/f78ce2d0ed32b58973518623/dragons/metadata.rb:1\nFC065: Ensure source_url is set in metadata: /var/folders/py/gk9z_67s0mjbg0w5m3xl_w180000gn/T/f78ce2d0ed32b58973518623/dragons/metadata.rb:1\nRun with Foodcritic Version 7.0.1 with tags ~FC031 ~FC045 and failure tags any",
    created_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00,
    updated_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00>,
   #<MetricResult:0x007fa8825ee098
    id: 3,
    cookbook_version_id: 27,
    quality_metric_id: 2,
    failure: true,
    feedback: "Failure: Cookbook has 0 collaborators. A cookbook must have at least 2 collaborators to pass this metric.",
    created_at: Tue, 15 Nov 2016 16:44:14 UTC +00:00,
    updated_at: Tue, 15 Nov 2016 16:44:14 UTC +00:00>],
 true=>
  [#<MetricResult:0x007fa8825ee8b8
    id: 2,
    cookbook_version_id: 27,
    quality_metric_id: 3,
    failure: false,
    feedback: "",
    created_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00,
    updated_at: Tue, 15 Nov 2016 16:44:13 UTC +00:00>]}
```